platform/ios/MGLMapView.mm

```diff
@@ -202,6 +203,9 @@ - (void)setStyleURL:(nullable NSURL *)styleURL
 - (void)commonInit
 {
     _isTargetingInterfaceBuilder = NSProcessInfo.processInfo.mgl_isInterfaceBuilderDesignablesAgent;
+    
+    std::unique_ptr<MGLLogObserver> logObserver = std::make_unique<MGLLogObserver>(self);
+    mbgl::Log::setObserver(std::move(logObserver));
```

Comment:
`mbgl::Log` is a singleton, so this approach won’t scale to multiple `MGLMapView`s.